queries/systemtap/highlights.scm

```diff
@@ -0,0 +1,156 @@
+(identifier) @variable
+
+(preprocessor_macro_definition
+  name: (identifier) @define)
+
+(preprocessor_macro_expansion) @function.macro
+
+(preprocessor_constant) @constant.macro
+
+(number) @number
+(string) @string
+(escape_sequence) @string.escape
+
+[
+  (script_argument_string)
+  (script_argument_number)
+] @constant
+
+(probe_point_component) @function
+
+(function_definition
+  name: (identifier) @function)
+
+(parameter
+  name: (identifier) @parameter)
+
+(type) @type.builtin
+
+(aggregation_operator) @function.macro
+
+(member_expression
+  member: (identifier) @field)
+
+(call_expression
+  function: (identifier) @function.call)
+
+(
+  (call_expression
+    function: (identifier) @function.builtin)
+  (#any-of? @function.builtin
+    "print" "printd" "printdln" "printf" "println"
+    "sprint" "sprintd" "sprintdln" "sprintf" "sprintln")
+)
+
+(
+  (identifier) @variable.builtin
+  (#lua-match? @variable.builtin "^\$+[0-9A-Z_a-z]+\$*$")
+)
+
+(shebang_line) @preproc
+
+(comment) @comment @spell
+
+[
+  "!"
+  "!="
+  "!~"
+  "$"
+  "$$"
+  "%"
+  "%="
+  "&"
+  "&&"
+  "&="
+  "*"
+  "*="
+  "+"
+  "++"
+  "+="
+  "-"
+  "--"
+  "-="
+  "->"
+  "."
+  ".="
+  "/"
+  "/="
+  ":"
+  "<"
+  "<<"
+  "<<<"
+  "<<="
+  "<="
+  "="
+  "=="
+  "=~"
+  ">"
+  ">="
+  ">>"
+  ">>="
+  "?"
+  "^"
+  "^="
+  "|"
+  "|="
+  "||"
+  "~"
+] @operator
+
+[
+  ","
+  (null_statement)
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+[
+  "break"
+  "continue"
+  "delete"
+  "global"
+  "limit"
+  "next"
+  "private"
+  "probe"
+] @keyword
+
+"function" @keyword.function
+"in" @keyword.operator
+"return" @keyword.return
+
+[
+  "if"
+  "else"
+] @conditional
+
+[
+  "for"
+  "foreach"
+  "while"
+] @repeat
+
+[
+  "try"
+  "catch"
+] @exception
+
+[
+  "%("
+  "%)"
+  "%:"
+  "%?"
+  "%{"
+  "%}"
```

Comment:
Why not `@punctuation.bracket`?

Reply:
The tokens `%(`, `%)`, `%:` and `%?` are meaningful only to the preprocessor, being interpreted in a separate pass before the SystemTap script gets parsed.

The delimiters `%{` and `%}` are used only for embedded C code, which is transcribed verbatim.

Another reason for choosing distinct captures for these tokens is to achieve consistency with the behavior of the language server.